team-validator.js

```diff
@@ -819,6 +820,9 @@ class Validator {
 							// must be able to breed with father
 							if (!dexEntry.eggGroups.some(eggGroup => eggGroupsSet.has(eggGroup))) continue;
 
+							let moveGen = (dexEntry.learnset[moveid] ? dexEntry.learnset[moveid][0].charAt(0) : dexEntry.learnset['sketch'][0].charAt(0));
+							if (noPastGen && (dexEntry.tier.indexOf('Bank') >= 0 || moveGen !== learnedGen.toString() || template.id === dexEntry.prevo || dexEntry.evos.indexOf(template.id) >= 0)) continue;
```

Comment:
`learnedGen.toString()` should probably be `'' + learnedGen`

For that matter, I'm pretty sure that if `noPastGen` is `true`, `learnedGen` is guaranteed to be `tools.gen` i.e. `7`.